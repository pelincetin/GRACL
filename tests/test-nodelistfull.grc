void printNodeList(NodeList nl) {
    /*
    for (Node n in nl) {
        print(n.data());
    }*/
    int i = nl.length_NL();
    while (i > 0){
        Node n = nl.removeFirst_NL();
        print(n.data());
        nl.appendNode(n);
        i = i - 1;
    }
}


int main()
{
    Graph g = createGraph(5);
    String greeting = "Hello";
    String nodedata1 = "noded1";
    String nodedata2 = "noded2";
    Node n1 = createNode(g, nodedata1);
    Node n2 = createNode(g, nodedata2);
    Node n3 = createNode(g, greeting);  
    bool success = True;

    NodeList nl = createNodeList();
    if (!nl.empty_NL()) {
        success = False;
    }

    nl.appendNode(n1);
    if (!(nl.length_NL() == 1)) {
        success = False;
    }
    nl.prependNode(n2);
    if (!(nl.length_NL() == 2)) {
        success = False;
    }
    if (!nl.includesNode(n2)) {
        success = False;
    }
    if (!nl.includesNode(n1)) {
        success = False;
    }

    nl.removeFirst_NL();
    if (!nl.includesNode(n1)) {
        success = False;
    }
    if (nl.includesNode(n2)) {
        success = False;
    }

    nl.prependNode(n2);
    nl.removeLast_NL();   
    if (!nl.includesNode(n2)) {
        success = False;
    }
    if (nl.includesNode(n1)) {
        success = False;
    }

    nl.appendNode(n1);
    nl.appendNode(n3);
    nl.removeNode(n1);
    nl.removeNode(n3);
    nl.removeNode(n2);
    if (!nl.empty_NL()) {
        success = False;
    }

    if (success) {
        print("SUCCESS");
    }
    else {
        print("FAILURE");
    }

    return 0;
}