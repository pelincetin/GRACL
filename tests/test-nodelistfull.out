Should be true: 
true
Append node 'hello'
Prepend node 'goodbye'
List should be goodbye hello: 
Goodbye
Hello
Data of head should be goodbye: 
Goodbye
Data of tail should be hello: 
Hello
Length should be 2: 
2
After remove first (list should be hello): 
Hello
Prepend node 'goodbye'
After remove last (list should be goodbye): 
Goodbye
Append node 'hello'
After remove node hello (list should be goodbye): 
Goodbye